templates/style.scss

```diff
@@ -84,21 +84,33 @@ div.rustdoc {
         }
     }
 
-    // this is actual fix for docs.rs navigation and rustdoc sidebar
-    position: fixed;
-    top: $top-navbar-height;
-    bottom: 0;
-    left: 0;
-    right: 0;
-    display: block;
-    overflow-y: auto;
+    position: relative;
+}
+
+html {
+    height: 100%;
 }
 
 body {
     padding: 0;
     margin: 0;
-    // Since top navbar is fixed, we need to add padding to the body content.
-    padding-top: $top-navbar-height;
+    height: 100%;
+    overflow: hidden;
+    margin: 0px;
+    display: flex;
+    box-sizing: border-box;
+}
+
+#page-container {
+    height: 100%;
+    width: 100%;
+    display: flex;
+    flex-direction: column;
```

Comment:
It'll reduce a bit the current amount of devices able to use docs.rs, but not too much. So I don't know. What do you think @jyn514 ?

Reply:
https://caniuse.com/#search=flex-direction shows that it isn't supported on Firefox, but I didn't notice anything wrong when I visited the site. @antifuchs what does this do? Can you maybe show screenshots before and after?

Reply:
The `flex-direction: column` is supposed to arrange child elements vertically below each other. I guess even if it's not supported, there is nowhere for the elements to flow to, since we're 100%ing them all.

Reply:
Then is it useful to keep it?

Reply:
maybe - I'll try removing it.

Reply:
Yep, turns out it's necessary: Webkit at least reflows elements to sit next to each other.
![screenshot showing the navbar on the left and the page body on the right](https://user-images.githubusercontent.com/11864/86398833-3afb5700-bc74-11ea-8547-5d2c18054807.png)

Reply:
Ok, if it's not a compatibility issue I don't see a problem. I'm still confused by the caniuse page but since it clearly works in my Firefox browser that page must have bad info.

Reply:
Agreed - I don't find supporting evidence on the MDN page that any `about:config` settings would be necessary, as claimed on the caniuse page; maybe the problem was referring to `-webkit-flex-direction`?

Reply:
My worry is mostly that most flex-based CSS properties were supported starting 2013-2014. So not that old (well, actually it's getting old now XD).

Reply:
Hah, I feel that worry - it definitely seems weirdly futuristic, even though it's 6 years old now & decently well-supported. From the [rust RFC on browser support](https://rust-lang.github.io/rfcs/1985-tiered-browser-support.html#supported-browsers), it does fit the support matrix (at least one firefox, at least one IE, two chromes, one of each for others); I guess this should make it fine?

Reply:
This is what we need to agree on. I'll let that decision up to the other members of the @rust-lang/docs-rs team decide. :p

Reply:
If it works on Android, Safari, and iOS, I'm all in favor.

Reply:
I couldn't test on Android (don't have an Android device here), but once a staging server works, it should hopefully be easy to enlist help from the community (: